lib/post_destroyer.rb

```diff
@@ -435,4 +427,19 @@ def update_post_counts(operator)
       end
     end
   end
+
+  def handle_post_flags
+    if permanent?
+      Reviewable.where(target: @post).destroy_all
```

Comment:
Does this destroy the current reviewable too? Don't we want to keep the current one?